doc/dependencies.txt

```diff
@@ -70,7 +70,7 @@ LLVM           | 10.0.0  | 13.0.0*     | Target-independent code generation
 Bison          | 3.0.0   | 3.7.0       | General-purpose parser generator                                  | Y       | Y        | https://www.gnu.org/software/gcc
 Flex           | 2.6.0   | 2.6.4       | Fast lexical analyzer generator                                   | Y       | Y        | https://github.com/westes/flex
 Python         | 3.9.1   | 3.10        | The python interpreter and libraries                              | Y       | Y        | https://www.python.org
-PyBind11       | 2.9.1   | Latest      | C++/python bindings                                               | Y       | Y        | https://pybind11.readthedocs.io
+nanobind       | 1.9.2   | Latest      | C++/python bindings                                               | Y       | Y        | https://nanobind.readthedocs.io
```

Comment:
Should we perhaps set the minimum version to 2.0.0? I don't believe we're testing 1.9.2 through the CI and given that this is a new dependency, perhaps it's better to stick to a newer version as our minimum?

Reply:
Sounds good to me. I'll make that change. Thanks for reviewing!